webapp-landing/src/main/java/fi/nls/oskari/RssFeedService.java

```diff
@@ -0,0 +1,122 @@
+    private static final String FEED_URL_NEWS = "http://www.maanmittauslaitos.fi/rss/paikkatietojen-yhteiskaytto";
+    private static int READ_INTERVAL = 60000  * 30; //30 minutes
+    private static int NUM_SHOW_ITEMS = 5;
+
+    private static final Logger logger = Logger.getLogger("RssFeedReader");
+
+    private RssFeedChannel notificationsChannel;
+    private RssFeedChannel newsChannel;
+    private List<RssFeedItem> notifications;
+    private List<RssFeedItem> news;
+    private HttpComponentsClientHttpRequestFactory clientHttpRequestFactory;
+    private ObjectMapper mapper;
+
+    public RssFeedService() {
+        clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
+        mapper = new ObjectMapper();
+        mapper.setDateFormat(new SimpleDateFormat("ddd, dd MMM yyyy HH:mm:SS Z"));
+        new Thread(this).start();
+    }
+
+    public List<RssFeedItem> getNotifications() {
+        return notifications;
+    }
+
+    public List<RssFeedItem> getNews() {
+        return news;
+    }
+
+    private RssFeedChannel readRssFeed(String feedUrl) {
+        try {
+            RssFeedChannel channel = null;
+            ClientHttpRequest req = clientHttpRequestFactory.createRequest(new URI(feedUrl), HttpMethod.GET);
+            ClientHttpResponse resp = req.execute();
+
+            if (resp.getStatusCode() == HttpStatus.OK) {
+                try {
+                    String xml = StreamUtils.copyToString(resp.getBody(), Charset.forName("UTF-8"));
+                    JSONObject jsonObject = XML.toJSONObject(xml);
+                    ObjectMapper objectMapper = new ObjectMapper();
+
+                    RssFeedXmlRoot rssFeedXmlRoot = objectMapper.readValue(jsonObject.toString(), RssFeedXmlRoot.class);
+                    channel = rssFeedXmlRoot.getRssFeed().getChannel();
+                }
+                catch (Exception ex) {
+                    logger.error("Couldn't deserialize RssFeedChannel " +feedUrl, ex);
+                }
+                return channel;
+            }
+            else {
+                logger.warn("RSS feed reader got status " + resp.getRawStatusCode() + "for url " + feedUrl);
+            }
+        }
+        catch (Exception ex) {
+            logger.error("Reading RSS feed failed", ex);
+        }
+        return null;
+    }
+
+    private void updateRssItemLists() {
+        notifications = pickLatestItems(notificationsChannel);
+        news = pickLatestItems(newsChannel);
+    }
+
+    /**
+     * Picks last X items from the feed. Presumes that items are in chronological order.
+     *
+     * @param channel
+     * @return
+     */
+    private List<RssFeedItem> pickLatestItems(RssFeedChannel channel) {
+        List<RssFeedItem> list = new Vector<RssFeedItem>();
+        if (channel != null && channel.getItems() != null) {
+            Collections.sort(channel.getItems());
+            for (int i = 0; i < channel.getItems().size() && i < NUM_SHOW_ITEMS; i++) {
```

Comment:
Instead of looping could use List#subList(int fromIndex, int toIndex); where toIndex would be the minimum between NUM_SHOW_ITEMS and the size of the list.

Reply:
Now using subList instead.